python/pandemic_simulator/environment/location/base_business.py

```diff
@@ -1,113 +1,70 @@
 # Confidential, Copyright 2020, Sony Corporation of America, All rights reserved.
-
-from typing import cast, Tuple, Optional, Type
-
-import numpy as np
+from abc import ABCMeta
+from typing import cast, Tuple, Type, TypeVar, ClassVar
 
 from .base import BaseLocation
-from ..interfaces import SimTime, SimTimeTuple, Registry, PersonID, LocationID, LocationRule, DEFAULT, \
+from ..interfaces import SimTime, SimTimeTuple, PersonID, LocationRule, DEFAULT, \
     BusinessLocationRule, BusinessLocationState, NonEssentialBusinessLocationState, \
     NonEssentialBusinessLocationRule
-from ...utils import checked_cast
 
 __all__ = ['BusinessBaseLocation', 'EssentialBusinessBaseLocation', 'NonEssentialBusinessBaseLocation',
            'AgeRestrictedBusinessBaseLocation']
 
+_BusinessState = TypeVar('_BusinessState', bound=BusinessLocationState)
+_NonEssentialBusinessState = TypeVar('_NonEssentialBusinessState', bound=NonEssentialBusinessLocationState)
+
 
-class BusinessBaseLocation(BaseLocation):
+class BusinessBaseLocation(BaseLocation[_BusinessState], metaclass=ABCMeta):
```

Comment:
This only partially implements the abstract classes, that is why it still has the metaclass. Otherwise, we would need to raise NotImplementedError everywhere.

Reply:
Maybe I can move this to interfaces?